locations/spiders/marriott_hotels.py

```diff
@@ -1,122 +1,81 @@
-
-        properties = {
-            "ref": re.search(r".*/(.*)/$", response.url).groups()[0],
-            "name": name,
-            "addr_full": street_address,
-            "city": city,
-            "state": state,
-            "postcode": response.xpath(
-                '//span[@itemprop="postalCode"]/text()'
-            ).extract_first(),
-            "country": response.xpath(
-                '//span[@itemprop="addressCountry"]/text()'
-            ).extract_first(),
-            "phone": (
-                response.xpath('//span[@itemprop="telephone"]/text()').extract_first()
-                or ""
-            ).strip("| "),
-            "lat": float(
-                response.xpath('//span[@itemprop="latitude"]/text()').extract_first()
-            ),
-            "lon": float(
-                response.xpath('//span[@itemprop="longitude"]/text()').extract_first()
-            ),
-            "website": response.url,
-            "brand": brand,
-        }
-
-        yield GeojsonPointItem(**properties)
-
-    def parse_ritz(self, response):
-        brands = {
-            "upcoming": "Ritz-Carlton",
-            "reserve": "Ritz-Carton Reserve",
-        }
-        data = re.search(
-            r"trc.pageProperties.trcMap.mapData= (.*)", response.text
-        ).groups()[0]
-        data = json.loads(data.strip(";\r\n "))
-
-        for item in data["response"]["list"]["listItems"]["items"]:
-            if item["venue"]["type"] in ("ritz", "upcoming"):
-                name = "The Ritz-Carlton " + item["venue"]["name"]
-            else:
-                name = item["venue"]["name"]
-            phone = re.split(r"<br./>", item["tip"]["text"])[-1]
-            properties = {
-                "ref": "-".join(
-                    re.search(r".*/(.*)/(.*)$", item["tip"]["link"]["url"]).groups()
-                ),
-                "name": name,
-                "addr_full": item["venue"]["location"]["address"].strip(),
-                "city": item["venue"]["location"]["city"].strip(),
-                "state": item["venue"]["location"]["state"].strip(),
-                "postcode": item["venue"]["location"]["postalCode"].strip(". "),
-                "country": item["venue"]["location"]["country"].strip(),
-                "phone": phone.strip(),
-                "lat": float(item["venue"]["location"]["lat"]),
-                "lon": float(item["venue"]["location"]["lng"]),
-                "website": item["tip"]["link"]["url"],
-                "brand": brands.get(item["venue"]["type"], "Ritz-Carlton"),
-            }
-
-            yield GeojsonPointItem(**properties)
-
-    def parse(self, response):
-        xml = Selector(response)
-        xml.remove_namespaces()
-
-        urls = xml.xpath("//loc/text()").extract()
-        for url in urls:
-            yield scrapy.Request(response.urljoin(url), callback=self.parse_hotel)
+        letters = "ABCDEFGHIJKLMNOPQRSTUVXYZ"
```

Comment:
Instead of making 676 requests for all combinations of the alphabet, can we use the `my_brands` dict that you put together and make requests for just the brands we know exist?

Reply:
You could and that is what my code used to do until my interest was raised when I ported it and found additional codes. Future movements in the endpoint would show themselves as ERROR logs. Otherwise we will never see them. It's perhaps worth a comment in the code. Your call of course.

Reply:
@iandees if you tell me which way you wish me to go on this I will do it.

Reply:
I think it's ok. Especially with the download_delay 👍🏻